Sources/StreamChat/WebSocketClient/EventMiddlewares/ChannelPinnedMessagesUpdaterMiddleware.swift

```diff
@@ -3,3 +3,83 @@
 //
 
 import Foundation
+
+struct ChannelPinnedMessagesUpdaterMiddleware<ExtraData: ExtraDataTypes>: EventMiddleware {
+    let database: DatabaseContainer
+
+    func handle(event: Event, completion: @escaping (Event?) -> Void) {
+        if let event = event as? MessageUpdatedEvent<ExtraData> {
+            let cid = event.cid
+            let messageId = event.messageId
+            handlePinnedMessages(
+                event: event,
+                cid: cid,
+                messageId: messageId,
+                handle: { (channel, message) in
+                    if message.pinned {
+                        channel.pinnedMessages.insert(message)
+                    } else {
+                        channel.pinnedMessages.remove(message)
+                    }
+                },
+                completion: completion
+            )
+        } else if let event = event as? MessageDeletedEvent<ExtraData> {
+            let cid = event.cid
+            let messageId = event.messageId
+            handlePinnedMessages(
+                event: event,
+                cid: cid,
+                messageId: messageId,
+                handle: { (channel, message) in
+                    if message.pinned {
+                        channel.pinnedMessages.remove(message)
+                    }
+                },
+                completion: completion
+            )
```

Comment:
This is probably not needed. Ideally, we should approach this as any other message -> the message is still visible, but it's marked deleted.